### PR TITLE
documentation #768: updated swagger for auth endpoints

### LINKED ISSUE #768: Update Swagger doc for Auth (2)
We are updating the documentation for all endpoints and adding more _description and details_. 

One of the main controllers is Auth. You should update the documentation according to the **pattern** below:
```ts
  @ApiBearerAuth()
  @ApiOkResponse({
    type: DisciplineTeachersResponse,
  })
  @ApiBadRequestResponse({
    description: `\n
    InvalidQueryException:
      Type of discipline must be a field of enum
    InvalidDisciplineIdException:
      Discipline with such id is not found`,
  })
  @ApiUnauthorizedResponse({
    description: `\n
    UnauthorizedException:
      Unauthorized`,
  })
  @ApiForbiddenResponse({
    description: `\n
    NoPermissionException:
      You do not have permission to perform this action`,
  })
  @ApiParam({
    name: 'disciplineId',
    required: true,
    description: 'Id of certain discipline',
  })
  @ApiQuery({
    name: 'disciplineType',
    required: true,
    enum: DisciplineTypeEnum,
    description: 'Discipline type of some discipline',
  })
  @Access(PERMISSION.GROUPS_$GROUPID_DISCIPLINES_TEACHERS_GET, GroupByDisciplineGuard)
  @Get('/:disciplineId/teachers')
  async getAllByDiscipline (
    @Param('disciplineId', DisciplineByIdPipe) disciplineId: string,
    @Query('disciplineType') disciplineType: DisciplineTypeEnum
  ) {
    const teachers = await this.disciplineService.getTeachers(disciplineId, disciplineType);
    return { teachers };
  }
```
**Structure:**
- auth (if needed)
- ok response
- client errors
- params/query description
- access (if needed)

You should also add **field description** in the bady or response.
Add **summary** to endpoints (fix this issue).
Set **default values** for the optional fields

image

_My DM is always open for you. Happy coding_ 🐸

[Our Swagger](https://apidev.fictadvisor.com/api)
[Doc](https://docs.nestjs.com/openapi/introduction)

### PR DESCRIPTION
**Pull request by SharpDevOps10**
- updated /me
- updated /forgotPassword
- updated /resetPassword(token)
- updated /verifyEmail
- updated /verifyEmail(token)
- updated /verifyIsRegistered

closes #768

### CODE REVIEW

**Review by illiakroshka — CHANGES_REQUESTED** [review]

**Review by illiakroshka — APPROVED** [review]

File: src/v2/api/dtos/VerificationEmailDTO.ts [hunk 1]
```diff
@@ -3,7 +3,9 @@ import { validationOptionsMsg } from '../../utils/GLOBALS';
 import { ApiProperty } from '@nestjs/swagger';
 
 export class VerificationEmailDTO {
-  @ApiProperty()
+  @ApiProperty({
+    description: 'An email of the user for whom resending a letter with confirmation of email address is requested',
+  })
   @IsEmail({}, validationOptionsMsg('Email is not email'))
     email: string;
```
Comment (aim4ik11) [inline src/v2/api/dtos/VerificationEmailDTO.ts:10]: Add IsNotEmpty decorator for email and appropriate exception
image

**Review by aim4ik11 — CHANGES_REQUESTED** [review]

**Review by aim4ik11 — APPROVED** [review]

File: src/v2/api/dtos/IdentityQueryDTO.ts [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
+import { ApiProperty } from '@nestjs/swagger';
 
 export class IdentityQueryDTO {
+  @ApiProperty({
+    description: 'The username for checking the registration',
+  })
   @Matches(new RegExp('^[' + ENG_REGEX + NUM_REGEX + '_' + ']{2,40}$'),
     validationOptionsMsg('Username is not correct (a-zA-Z0-9_), or too short (min: 2), or too long (max: 40)'))
   @IsNotEmpty(validationOptionsMsg('Username is empty'))
     username?: string;
```
Comment (fokaaas) [inline src/v2/api/dtos/IdentityQueryDTO.ts:12]: Make it required

File: src/v2/api/dtos/IdentityQueryDTO.ts [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
 
+  @ApiProperty({
+    description: 'The email for checking the registration',
+  })
   @IsEmail({}, validationOptionsMsg('Email is not an email'))
   @IsNotEmpty(validationOptionsMsg('Email is empty'))
-    email?:string;
+    email?: string;
```
Comment (fokaaas) [inline src/v2/api/dtos/IdentityQueryDTO.ts:19]: Make it required

File: src/v2/api/responses/VerifyEmailTokenResponse.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { ApiProperty } from '@nestjs/swagger';
+
+export class VerifyEmailTokenResponse {
```
Comment (fokaaas) [inline src/v2/api/responses/VerifyEmailTokenResponse.ts:3]: It is JWTTokensResponse

**Review by fokaaas — CHANGES_REQUESTED** [review]

**Review by fokaaas — APPROVED** [review]

**Pull request merged**